### ISSUE TITLE
Add high score menu for MCA games(Unsolved)

### ISSUE THREAD

**Issue by ryanjames1729** [Vegz78/McAirpos#38]
Feel free to discard this, but would it be possible to add a game in the menu that loads the high scores for the all the games? I feel like it would be possible reading the data folders for each game. Trying to avoid creating a separate app outside of the Pi for this.

**Comment by Vegz78** [comment]
Hi @ryanjames1729,

Do I understand your idea correctly as a game programmed in MakeCode Arcade(MCA) that could read and display the highscores from other MCA games?

Interesting idea, but I am afraid I personally neither know all options inside MCA for file reading outside one game’s own data files nor enough about the internals of the pxt engine to claim for sure that it would be possible with a custom extension(though probably).

To be honest, I don’t know that much about MCA outside the work others and I did for getting the games to run and exit, as well as getting working support for gamepads on the RPi. I have made some futile attempts to understand the uf2 file format better, with the aim of running MCA games on the RPi directly from an exported/compiled uf2 file(which can be loaded directly back into the web editor) instead of the RAW elf executable, but I am still stuck.

Though, it should certainly be possible somehow to, as you write, “be possible reading the data folders for each game”, but I just don’t know how or the best way myself, unfortunately.

If it could help you somehow to keep the issue open for a while here, I will, but you might be more lucky on the official MCA forum or the Microsoft pxt or pxt-arcade GitHub pages. If you are lucky, you will reach and get helpful response from one of the
skillfull superusers or Microsoft tech people there.
